traces_profiler.go

```diff
@@ -0,0 +1,70 @@
+package sentry
+
+func (span *Span) startProfiling() {
+	var sampleRate = span.clientOptions().ProfilesSampleRate
+	switch {
+	case sampleRate < 0.0 || sampleRate > 1.0:
+		Logger.Printf("Skipping transaction profiling: ProfilesSampleRate out of range [0.0, 1.0]: %f", sampleRate)
+	case sampleRate == 0.0 || rng.Float64() >= sampleRate:
+		Logger.Printf("Skipping transaction profiling: ProfilesSampleRate is: %f", sampleRate)
+	default:
+		span.profiler = &_transactionProfiler{
+			stopFunc: startProfiling(),
+		}
+	}
+}
+
+type transactionProfiler interface {
+	Finish(span *Span, event *Event) *profileInfo
+}
+
+type _transactionProfiler struct {
```

Comment:
I know naming is hard but maybe we can rename the interface to `Profiler` and then call the implementation `transactionProfiler`.

Also, why not put it in its own package?

Reply:
> I know naming is hard but maybe we can rename the interface to `Profiler` and then call the implementation `transactionProfiler`.

The idea behind the `transactionProfiler` name is that it wraps the generic profiler functionality (defined in `profiler.go`) that doesn't have any relation to transactions, and uses it to profile transactions.

> Also, why not put it in its own package?

Actually, I started in a separate internal package for `profiler.go` but had to move to the root dir because of references to types in the root package (`Frame`, `DebugMeta`, although the latter is actually unused).
Or did you mean something else?